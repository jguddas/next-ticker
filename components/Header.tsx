import React from 'react'
import { x } from '@xstyled/styled-components'

import Text from './Text'

type Props = {
  children: React.ReactNode
  subHeader?: string
}

export default ({ children, subHeader }: Props): JSX.Element => (
  <x.div mb="56px">
    <x.h1 color="heading" fontSize="heading" lineHeight="heading">
      {children}
    </x.h1>
    {subHeader ? <Text>{subHeader}</Text> : null}
  </x.div>
)
