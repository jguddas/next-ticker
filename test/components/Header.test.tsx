import React from 'react'
import { render } from '../testUtils'
import Header from '../../components/Header'
import 'jest-styled-components'

describe('Header', () => {
  it('matches snapshot', () => {
    const { asFragment } = render(<Header>Hello World</Header>, {})
    expect(asFragment()).toMatchSnapshot()
  })
})
