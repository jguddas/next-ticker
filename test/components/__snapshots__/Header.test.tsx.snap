// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header matches snapshot 1`] = `
<DocumentFragment>
  .c0.c0 {
  margin-bottom: 56px;
}

.c1.c1 {
  color: heading;
  font-size: heading;
  line-height: heading;
}

<div
    class="c0"
  >
    <h1
      class="c1"
    >
      Hello World
    </h1>
  </div>
</DocumentFragment>
`;

exports[`Header with subHeader matches snapshot 1`] = `
<DocumentFragment>
  .c0.c0 {
  margin-bottom: 56px;
}

.c2.c2 {
  color: text;
  font-size: 1px;
}

.c1.c1 {
  color: heading;
  font-size: heading;
  line-height: heading;
}

<div
    class="c0"
  >
    <h1
      class="c1"
    >
      Hello World
    </h1>
    <div
      class="c2"
    >
      â€¦
    </div>
  </div>
</DocumentFragment>
`;
